### ISSUE TITLE
What base model is the training script using?

### ISSUE THREAD

**Issue by w32zhong** [SafeAILab/EAGLE#181]
Hello, thank you again for such great work. I would like to check my understanding here because I am confused by the differences between training  and inference scripts.

In the training data generation, a local model directory is loaded. See https://github.com/SafeAILab/EAGLE/blob/v1/eagle/ge_data/ge_data_all_llama2chat.py#L23.

I used to assume this loaded model is the same as the hgf llama2-chat model, in this case, [when you read last-layer hidden state](https://github.com/SafeAILab/EAGLE/blob/v1/eagle/ge_data/ge_data_all_llama2chat.py#L185C43-L186), it should have been (layer-)normalized according to [the official hgf llama implementation](https://github.com/huggingface/transformers/blob/2ab7bdc40333b230b642f09e8334fb8e1a92d2a4/src/transformers/models/llama/modeling_llama.py#L611-L615). 

However, I have recently discovered that your inference code does not do layer normalization (and also `input_layernorm`) in draft decoder layer. I.e., during speculative decoding, when you [read the top hidden states of base model](https://github.com/SafeAILab/EAGLE/blob/v1/eagle/model/cnets.py#L775-L777), it [no longer does any final-layer normalization](https://github.com/SafeAILab/EAGLE/blob/v1/eagle/model/cnets.py#L630-L638).

In order to match the training and inference, I guess the local model being loaded before training (e.g., "/home/lyh/weights/hf/llama/7B/") is a customized llama2 model as well? (e.g., using `auto_map` in config.json). Is my understanding correct? Thanks in advance.

**Comment by w32zhong** [comment]
Sorry, I think I've mistakenly think that the draft decoder has to match base model architecture. I guess it was just trained this way to predict top hidden states without using layer normalization (and `input_layernorm`). I will close this issue unless you want to correct my understanding.
